src/auth/oauth2common.ts

```diff
@@ -160,45 +160,44 @@ export abstract class OAuthClientAuthHandler {
   private injectAuthenticatedRequestBody(opts: GaxiosOptions) {
     if (this.#clientAuthentication?.confidentialClientType === 'request-body') {
       const method = (opts.method || 'GET').toUpperCase();
-      // Inject authenticated request body.
-      if (METHODS_SUPPORTING_REQUEST_BODY.indexOf(method) !== -1) {
-        // Get content-type.
-        let contentType;
-        const headers = opts.headers || {};
-        for (const key in headers) {
-          if (key.toLowerCase() === 'content-type' && headers[key]) {
-            contentType = headers[key].toLowerCase();
-            break;
-          }
-        }
-        if (contentType === 'application/x-www-form-urlencoded') {
-          opts.data = opts.data || '';
-          const data = querystring.parse(opts.data);
-          Object.assign(data, {
-            client_id: this.#clientAuthentication!.clientId,
-            client_secret: this.#clientAuthentication!.clientSecret || '',
-          });
-          opts.data = querystring.stringify(data);
-        } else if (contentType === 'application/json') {
-          opts.data = opts.data || {};
-          Object.assign(opts.data, {
-            client_id: this.#clientAuthentication!.clientId,
-            client_secret: this.#clientAuthentication!.clientSecret || '',
-          });
-        } else {
-          throw new Error(
-            `${contentType} content-types are not supported with ` +
-              `${this.#clientAuthentication!.confidentialClientType} ` +
-              'client authentication'
-          );
-        }
-      } else {
+
+      if (!METHODS_SUPPORTING_REQUEST_BODY.includes(method)) {
```

Comment:
I see why you made this change - definitely more readable! - but ideally could be another PR (given this is a big PR already, and this is more of a refactor, and we're already doing so many other gaxios changes). NB.